src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,123 @@
+import classNames from 'classnames';
+import { FC, useMemo } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { Person } from '../../types';
+import { getVisiblePeople } from '../../utils/getVisiblePeople';
+import { PersonNavLink } from '../PersonNavLink';
+import { SortLink } from '../SortLink';
+
+type Props = {
+  people: Person[],
+  selectedSlug: string,
+};
+
+export const PeopleTable: FC<Props> = ({ people, selectedSlug }) => {
+  const [searchParams] = useSearchParams();
+  const centuries = searchParams.getAll('centuries');
+  const sexFilter = searchParams.get('sex');
+  const query = searchParams.get('query');
+  const sort = searchParams.get('sort') as keyof Person;
+  const isReversed = searchParams.get('order') === 'desc';
+
+  const preparedPeople = useMemo(() => (
+    getVisiblePeople(people, centuries, sexFilter, query, sort, isReversed)
+  ), [people, centuries, sexFilter, query, sort, isReversed]);
```

Comment:
I checked my previous comment, and found that your solution was greater than the current one
With 2 separate functions for filter and sort functionality we can avoid array filtering one more time when only sort value changes